core/src/test/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactoryTest.java

```diff
@@ -73,6 +77,32 @@ public void initialize(TelemetryConfiguration configuration) {
     }
 
     @Test
+    public void configurationWithNullIkeyTest() {
```

Comment:
Where's the code for trimming?
Also, what about the case that there is a valid iKey preceded/followed by spaces?